securedrop_client/storage.py

```diff
@@ -384,7 +390,7 @@ def __update_submissions(
 
     for submission in remote_submissions:
         local_submission = local_submissions_by_uuid.get(submission.uuid)
-        if local_submission and submission.source_uuid not in skip_uuids_deleted_conversation:
+        if local_submission:
```

Comment:
If there is still a `local_submission` object returned, then interact with it normally. If the submission was locally deleted then the item will be treated like a 'new' submission, not a submission to update.